### PR TITLE
Add examples to `Time::Format` methods

### LINKED ISSUE #7472: Time format default constants
## Enhancement
Working with time in crystal can be a little difficult as [shown here](https://play.crystal-lang.org/#/r/6bwk). Sometimes it's hard to remember the exact format that was used, and how to actually write that out. 

For those familiar with using rails, there's a [neat date method](https://apidock.com/rails/Date/to_s) that allows you to specify a default format. You don't have to remember how to write `%Y-%m-%d %H:%M %p` and know what that means if all you really want is something like `:db`. Since crystal has an awesome way of converting symbols in to enums, this would be a great way to have something like

```crystal
t = Time.now.to_s(:db)
Time.parse_utct(t, :db)

t = Time.now.to_s(Time::Format::DB)
Time.parse_utc(t, Time::Format::DB)
```

There could be a set of the most commonly used formats setup, along with a way to extend that list for libraries that want to add in their own additional formats that may be useful but less common.

### PR DESCRIPTION
**Pull request by ThunderKey**
I've added examples of the different time formats to the formatters and the `Time#parse_*` functions.

If I can improve something let me know!

Fixes #7472

### CODE REVIEW

**Review by straight-shoota — APPROVED** [review]

File: src/time.cr [hunk 1]
```diff
@@ -1118,6 +1118,11 @@ struct Time
   end
 
   # Parse time format specified by [RFC 3339](https://tools.ietf.org/html/rfc3339) ([ISO 8601](http://xml.coverpages.org/ISO-FDIS-8601.pdf) profile).
+  #
+  # Example:
```
Comment (Sija) [inline src/time.cr:1122]: Redundant, ditto all below.

```suggestion
```

File: src/time.cr [hunk 2]
```diff
@@ -1128,6 +1133,11 @@ struct Time
   # In ISO 8601 for examples, field delimiters (`-`, `:`) are optional.
   #
   # Use `#to_rfc3339` to format a `Time` according to .
+  #
+  # Example:
```
Comment (Sija) [inline src/time.cr:1137]:
```suggestion
```

File: src/time.cr [hunk 3]
```diff
@@ -1154,6 +1164,11 @@ struct Time
   # Parse time format specified by [RFC 2822](https://www.ietf.org/rfc/rfc2822.txt).
   #
   # This is also compatible to [RFC 882](https://tools.ietf.org/html/rfc882) and [RFC 1123](https://tools.ietf.org/html/rfc1123#page-55).
+  #
+  # Example:
```
Comment (Sija) [inline src/time.cr:1168]:
```suggestion
```

File: src/time/format/custom/iso_8601.cr [hunk 4]
```diff
@@ -116,6 +116,13 @@ struct Time::Format
   end
 
   # The ISO 8601 date format.
+  #
+  # Example:
+  # ```
+  # Time::Format::ISO_8601_DATE.parse("2016-02-15") # => 2016-02-15 00:00:00.0 UTC
+  #
+  # Time::Format::ISO_8601_DATE.format(Time.utc(2016, 2, 15, 4, 35, 50)) # => "2016-02-15"
+  # ```
```
Comment (Sija) [inline src/time/format/custom/iso_8601.cr:125]:
```suggestion
  # ```
  # Time::Format::ISO_8601_DATE.parse("2016-02-15")                      # => 2016-02-15 00:00:00.0 UTC
  # Time::Format::ISO_8601_DATE.format(Time.utc(2016, 2, 15, 4, 35, 50)) # => "2016-02-15"
  # ```
```

File: src/time/format/custom/iso_8601.cr [hunk 5]
```diff
@@ -140,6 +147,13 @@ struct Time::Format
   end
 
   # The ISO 8601 date time format.
+  #
+  # Example:
+  # ```
+  # Time::Format::ISO_8601_DATE_TIME.format(Time.utc(2016, 2, 15, 4, 35, 50)) # => "2016-02-15T04:35:50Z"
+  #
+  # Time::Format::ISO_8601_DATE_TIME.parse("2016-02-15T04:35:50Z") # => 2016-02-15 04:35:50.0 UTC
+  # ```
```
Comment (Sija) [inline src/time/format/custom/iso_8601.cr:156]:
```suggestion
  # ```
  # Time::Format::ISO_8601_DATE_TIME.format(Time.utc(2016, 2, 15, 4, 35, 50)) # => "2016-02-15T04:35:50Z"
  # Time::Format::ISO_8601_DATE_TIME.parse("2016-02-15T04:35:50Z")            # => 2016-02-15 04:35:50.0 UTC
  # ```
```

File: src/time/format/custom/iso_8601.cr [hunk 6]
```diff
@@ -164,6 +178,13 @@ struct Time::Format
   end
 
   # The ISO 8601 time format.
+  #
+  # Example:
+  # ```
+  # Time::Format::ISO_8601_TIME.format(Time.utc(2016, 2, 15, 4, 35, 50)) # => "04:35:50Z"
+  #
+  # Time::Format::ISO_8601_TIME.parse("04:35:50Z") # => 0001-01-01 04:35:50.0 UTC
+  # ```
```
Comment (Sija) [inline src/time/format/custom/iso_8601.cr:187]:
```suggestion
  # ```
  # Time::Format::ISO_8601_TIME.format(Time.utc(2016, 2, 15, 4, 35, 50)) # => "04:35:50Z"
  # Time::Format::ISO_8601_TIME.parse("04:35:50Z")                       # => 0001-01-01 04:35:50.0 UTC
  # ```
```

File: src/time/format/custom/rfc_2822.cr [hunk 7]
```diff
@@ -2,6 +2,14 @@ struct Time::Format
   # The [RFC 2822](https://tools.ietf.org/html/rfc2822) datetime format.
   #
   # This is also compatible to [RFC 882](https://tools.ietf.org/html/rfc882) and [RFC 1123](https://tools.ietf.org/html/rfc1123#page-55).
+  #
+  # Example:
```
Comment (Sija) [inline src/time/format/custom/rfc_2822.cr:6]:
```suggestion
```

File: src/time/format/custom/rfc_3339.cr [hunk 8]
```diff
@@ -1,5 +1,13 @@
 struct Time::Format
   # The [RFC 3339](https://tools.ietf.org/html/rfc3339) datetime format ([ISO 8601](http://xml.coverpages.org/ISO-FDIS-8601.pdf) profile).
+  #
+  # Example:
```
Comment (Sija) [inline src/time/format/custom/rfc_3339.cr:4]:
```suggestion
```

File: src/time.cr (hunk 1 above)
Comment (ThunderKey) [reply src/time.cr]: That's a great idea. Thanks!

**Review by oprypin — APPROVED** [review]

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**